### DISCUSSION TITLE
Log.wien credentials

### DISCUSSION THREAD

**Discussion by webwude** [DarwinsBuddy/WienerNetzeSmartmeter#16] [General]
Hi.

I registered my smart meter yesterday, will be installed in two weeks. Log in on the portal is already possible, without any data.

When I try to log in with log.wien credentials through UI of the integration I got an error stating wrong username or password.

What am I missing?

**Comment by DarwinsBuddy** [comment]
I did Not use the login API of log.wien but the one Provider via the smart meter API.

See: https://github.com/DarwinsBuddy/WienerNetzeSmartmeter/issues/5#issuecomment-1256891621

try it out and report back :)

**Reply by webwude** [reply]
Unfortunately it seems that this is the only way to log in: if I open https://smartmeter-web.wienernetze.at/ I get the log in screen of log.wien and can log in with my credentials. Using the same in the integration with UI, I get "Username/password invalid".

I tried it via yaml as well. 
I tried it with my credentials and Zählpunkt according to the letter from Wiener Netze, but I get the error:
2022-11-11 09:16:55.957 ERROR (MainThread) [custom_components.wnsm.sensor] Error retrieving data from smart meter api.
Traceback (most recent call last):
File "/config/custom_components/wnsm/sensor.py", line 167, in async_update
zp = await self.get_zaehlpunkt(smartmeter)
File "/config/custom_components/wnsm/sensor.py", line 118, in get_zaehlpunkt
raise RuntimeError(f"Cannot access Zaehlpunkt {self.zaehlpunkt}")
RuntimeError: Cannot access Zaehlpunkt AT001000000000000000100001xxxxxx

Is it possible to check if it is a log in issue - or that the meter is not yet installed (and cannot deliver data)?

**Reply by webwude** [reply]
Today the smart meter was installed. As soon as I get data in the portal I will try again...

**Reply by ltaferner** [reply]
@webwude what did you do to login? 

I also get everytime I try to "Username/password invalid"

**Reply by webwude** [reply]
@ltaferner it worked out of nowhere all of the sudden.

**Comment by webwude** [comment]
It is working now - partly. I can enter my credentials without error - but not data (so far). 

![grafik](https://user-images.githubusercontent.com/7056632/203735414-0d7bab95-f843-4742-a0d9-bb656103e621.png)

**Reply by DarwinsBuddy** [reply]
I've heard from others that recently installed smartmeters take some time to deliver data.
Maybe it's not yet ready (see SmartMeterReady: false)

**Reply by DarwinsBuddy** [reply]
https://github.com/DarwinsBuddy/WienerNetzeSmartmeter/discussions/19

**Comment by webwude** [comment]
Login via Log.wien and data is working now, even though it still states "smart meter ready" false.

**Reply by DarwinsBuddy** [reply]
this is just the returned value of the API

Since there is no public documentation about it, we can only guess what that means.
though, I can reassure you: my sensor claims the same. seems like it's superfluous.

**Reply by webwude** [reply]
@DarwinsBuddy meanwhile even this value is true. You never know...

**Reply by DarwinsBuddy** [reply]
interesting 🤔
anyway I just took all values returned by the API and imported it into extra attributes of the hassio sensor.

I didn't intend to cause confusion.

**Comment by DarwinsBuddy** [comment]
closing due to staleness
